src/services/EcoBridge/Socket/Socket.utils.ts

```diff
@@ -0,0 +1,97 @@
+import { formatUnits } from '@ethersproject/units'
+import { ChainId } from '@swapr/sdk'
+import { isFee } from './Socket.types'
+import { Route, TokenPriceResponseDTO } from './api/generated'
+import { DAI_ARBITRUM_ADDRESS, DAI_ETHEREUM_ADDRESS } from '../../../constants'
+
+export const getDAIAddress = (chainId: ChainId) => {
+  switch (chainId) {
+    case ChainId.ARBITRUM_ONE:
+      return DAI_ARBITRUM_ADDRESS
+    case ChainId.MAINNET:
+      return DAI_ETHEREUM_ADDRESS
+    default:
+      return undefined
+  }
+}
+
+export const getBestRoute = (routes: Route[], tokenData?: TokenPriceResponseDTO, toTokenDecimals?: number) => {
+  if (routes.length === 1 || !tokenData || !toTokenDecimals) return routes[0]
+
+  const {
+    result: { tokenPrice }
+  } = tokenData
+
+  const bestRoute = routes.reduce<{ amount: number; routeId: string }>(
+    (total, txRoute) => {
+      if (tokenData?.success) {
+        const amount =
+          Number(formatUnits(txRoute.toAmount, toTokenDecimals).toString()) * tokenPrice - txRoute.totalGasFeesInUsd
+
+        const route = {
+          amount: Number(amount),
+          routeId: txRoute.routeId
+        }
+
+        //find better way to do it
+        if (route.amount <= 0) {
+          if (total.amount <= route.amount && total.amount !== 0) {
+            total = route
+            return total
+          }
+
+          total = route
+        } else {
+          if (total.amount <= route.amount) {
+            total = route
+          }
+        }
+
+        return total
+      }
+      return total
+    },
+    { amount: 0, routeId: '' } as { amount: number; routeId: string }
+  )
+
+  const indexOfBestRoute = routes.findIndex(route => route.routeId === bestRoute.routeId)
+
+  if (indexOfBestRoute === -1) return undefined
+
+  return routes[indexOfBestRoute]
+}
+
+export const getBridgeFee = (userTxs: any, fromAsset: { amount: string; decimals?: number }): string => {
+  if (isFee(userTxs)) {
+    const [singleTxBridge] = userTxs
+
+    //get protocolFee for each step
+    const totalStepsFee = singleTxBridge.steps.reduce((total, step) => {
+      if (!step.protocolFees.asset || step.protocolFees.amount === '0') {
+        return total
+      }
+      total += Number(formatUnits(step.protocolFees.amount, step.protocolFees.asset.decimals))
+      return total
+    }, 0)
+
+    if (totalStepsFee === 0) return '0%'
```

Comment:
why is this condition special? Can't it be dealt with the logic below?
It seems to me it does.
As a potential optimisation, how much is gained by doing this? If not much, maybe we should remove it

Reply:
Good catch. We will remove this line.

Reply:
Resolved by #900